### PR TITLE
Add options to silence logs and stdout from singer

### PR DESCRIPTION
**Pull request by ericboucher**
Adding two methods to make stdout more readable

1. Add a LOGLEVEL implementation for the tap logger
2. Create a simple filter to remove singer messages from stdout during tests

### CODE REVIEW

**Review by laurentS — APPROVED** [review]
There are a few issues in the sdk around logging:
https://gitlab.com/meltano/sdk/-/issues/170
https://gitlab.com/meltano/sdk/-/issues/386

maybe worth pushing your level config option to the sdk level.
And adding typing to the the Filtered output methods would be nice :)

File: tap_github/tap.py [hunk 1]
```diff
@@ -2,17 +2,34 @@
...
+import os
+import logging
+
 from singer_sdk import Stream, Tap
 from singer_sdk import typing as th  # JSON schema typing helpers
+from singer_sdk.helpers._classproperty import classproperty
 
 from tap_github.streams import Streams
 
+LOGLEVEL = os.environ.get("LOGLEVEL", "WARN").upper()
```
[the remark this answers is not in the document]
Comment (ericboucher) [reply tap_github/tap.py:14]: Should I move this to the config then?

[the remark this answers is not in the document]
Comment (ericboucher) [reply tap_github/tap.py:14]: Or both maybe? env is pretty common for LOGLEVEL

File: tap_github/tap.py [hunk 2]
```diff
@@ -2,17 +2,34 @@
...
 
+    @classproperty
+    def logger(cls) -> logging.Logger:
+        """Get logger.
+
+        Returns:
+            Logger with local LOGLEVEL.
+        """
+        logger = logging.getLogger(cls.name)
+        logger.setLevel(LOGLEVEL)
```
[the remark this answers is not in the document]
Comment (ericboucher) [reply tap_github/tap.py:32]: cc @edgarrmondragon

File: tap_github/tap.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ericboucher) [reply tap_github/tap.py]: Didn't find a good way to move this to the config since we are overriding a classproperty. Leaving it like that for now.

**Pull request merged**